### ISSUE TITLE
Getting "Alignment trap:" errors on arm.

### ISSUE THREAD

**Issue by JodiTheTigger** [Celtoys/Remotery#75]
As per
http://stackoverflow.com/questions/16548059/how-to-trap-unaligned-memory-access

enable signals on mis-aligned accesses

`echo 4 > /proc/cpu/alignment`

This gives a `SIGBUS` in `MessageQueue_AllocMessage` on the line
```c
msg->thread_sampler = thread_sampler;
```

MessageQueue_AllocMessage needs to allocate lengths that are 4 byte aligned.

I fixed it by adding the lines:
```c
    // needs to be 4 byte aligned on ARM
    payload_size = ((payload_size + 3) & ~3u);
```

**Comment by dwilliamson** [comment]
Thanks for spotting this. I think the fix should modify `write_size` on [line 2899](https://github.com/Celtoys/Remotery/blob/master/lib/Remotery.c#L2899).

I'll fix it soon as I have a bunch of stuff to get through first.

**Comment by JodiTheTigger** [comment]
I did that initially, but then got a crash on [Line 2985](https://github.com/Celtoys/Remotery/blob/master/lib/Remotery.c#L2985) so that's why I changed the actual payload size.

**Comment by dwilliamson** [comment]
Right. That makes sense!

**Comment by JodiTheTigger** [comment]
Ugh, if I apply the `payload_size` fix, then `ENL_SAMPLE_LOG_TEXT()` breaks on me. Goddammit.

Please help me, I have NFI what I'm doing now :-(

**Comment by JodiTheTigger** [comment]
Ok, I opened my third draw down, got a clue out of it, and figured it out.

You're correct. I had to modify `write_size` on [line 2889](https://github.com/Celtoys/Remotery/blob/master/lib/Remotery.c#L2899)

What was missing was that I _also_ had to update `message_size` on [line 2984](https://github.com/Celtoys/Remotery/blob/master/lib/Remotery.c#L2984) as well.

**Comment by dwilliamson** [comment]
`payload_size` is used to [send the log text size](https://github.com/Celtoys/Remotery/blob/master/lib/Remotery.c#L3892) so it's possible the string read on the web view side crashes out with invalid characters.

I haven't got a machine with the latest branch right now so can't completely test any patches yet. Will do soon.

**Comment by dwilliamson** [comment]
Fixed in b2fd9cbc33dff5e4cef145b65546211f4d543d73
